File: app/src/main/java/com/company/planet/ui/screens/HomeScreen.kt
package com.company.planet.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.company.planet.data.Phone
import androidx.compose.animation.AnimatedContent
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.slideInVertically
import androidx.compose.animation.slideOutVertically
import androidx.compose.animation.togetherWith
import androidx.compose.animation.core.tween
import com.company.planet.ui.components.TppLogo
import com.company.planet.ui.components.tppCard3D
import com.company.planet.ui.components.tppDropIn
import com.company.planet.ui.components.TppTabBar
import com.company.planet.ui.components.TppToast
import com.company.planet.ui.theme.Bg
import com.company.planet.ui.theme.BgPanel
import com.company.planet.ui.theme.Danger
import com.company.planet.ui.theme.Line
import com.company.planet.ui.theme.TextPrimary
import com.company.planet.ui.theme.TextSecondary
import com.company.planet.ui.util.rememberResponsive
import com.company.planet.viewmodel.AppScreen
import com.company.planet.viewmodel.PhoneViewModel

@Composable
fun HomeScreen(
    viewModel: PhoneViewModel,
    modifier: Modifier = Modifier
) {
    val state by viewModel.uiState.collectAsState()
    val responsive = rememberResponsive()
    var deleteTarget by remember { mutableStateOf<Phone?>(null) }

    Box(
        modifier = modifier
            .fillMaxSize()
            .background(
                Brush.radialGradient(
                    colors = listOf(Color(0x0F5EEAD4), Color.Transparent),
                    radius = 900f,
                    center = androidx.compose.ui.geometry.Offset(0.15f, 0f)
                )
            )
            .background(
                Brush.radialGradient(
                    colors = listOf(Color(0x0DA78BFA), Color.Transparent),
                    radius = 900f,
                    center = androidx.compose.ui.geometry.Offset(1f, 0f)
                )
            )
            .background(Bg)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .statusBarsPadding()
                .navigationBarsPadding()
                .padding(horizontal = responsive.horizontalPadding)
                .widthIn(max = 1280.dp)
                .align(Alignment.TopCenter)
        ) {
            TopBar(
                currentScreen = state.currentScreen,
                onScreenSelected = { screen ->
                    val keepForm = screen == AppScreen.ADD && state.editingPhone != null
                    viewModel.navigateTo(screen, keepForm = keepForm)
                },
                compact = responsive.compactHeader
            )

            Spacer(Modifier.height(18.dp))

            Column(
                modifier = Modifier
                    .weight(1f)
            ) {
                AnimatedContent(
                    targetState = state.currentScreen,
                    transitionSpec = {
                        (fadeIn(tween(280)) + slideInVertically(tween(280)) { it / 8 })
                            .togetherWith(fadeOut(tween(180)) + slideOutVertically(tween(180)) { -it / 10 })
                    },
                    label = "screenTransition"
                ) { screen ->
                    when (screen) {
                        AppScreen.ADD -> AddPhoneScreen(
                            editingPhone = state.editingPhone,
                            viewModel = viewModel,
                            modifier = Modifier
                                .fillMaxSize()
                                .fillMaxWidth()
                        )
                        else -> Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .verticalScroll(rememberScrollState())
                        ) {
                            when (screen) {
                                AppScreen.DASHBOARD -> DashboardScreen(
                                    phones = state.phones,
                                    totals = state.totals,
                                    viewModel = viewModel
                                )
                                AppScreen.INVENTORY -> InventoryScreen(
                                    state = state,
                                    viewModel = viewModel,
                                    onDeleteRequest = { deleteTarget = it }
                                )
                                else -> Unit
                            }
                            Spacer(Modifier.height(80.dp))
                        }
                    }
                }
            }
        }

        state.previewPhone?.let { phone ->
            PhonePreviewSheet(
                phone = phone,
                onDismiss = viewModel::closePreview,
                onEdit = { viewModel.openEditForm(phone) },
                onDelete = { deleteTarget = phone },
                fullScreen = responsive.useFullScreenSheet
            )
        }

        deleteTarget?.let { phone ->
            AlertDialog(
                onDismissRequest = { deleteTarget = null },
                title = { Text("Delete phone?", color = TextPrimary) },
                text = { Text("Delete \"${phone.model.ifBlank { "this phone" }}\"? This cannot be undone.", color = TextSecondary) },
                confirmButton = {
                    TextButton(onClick = {
                        viewModel.deletePhone(phone)
                        deleteTarget = null
                    }) { Text("Delete", color = Danger) }
                },
                dismissButton = {
                    TextButton(onClick = { deleteTarget = null }) { Text("Cancel") }
                },
                containerColor = BgPanel
            )
        }

        state.errorMessage?.let { message ->
            AlertDialog(
                onDismissRequest = viewModel::clearError,
                title = { Text("Error", color = TextPrimary) },
                text = { Text(message, color = TextSecondary) },
                confirmButton = { TextButton(onClick = viewModel::clearError) { Text("OK") } },
                containerColor = BgPanel
            )
        }

        TppToast(
            message = state.toastMessage,
            onDismiss = viewModel::clearToast,
            modifier = Modifier.fillMaxSize()
        )
    }
}

@Composable
private fun TopBar(
    currentScreen: AppScreen,
    onScreenSelected: (AppScreen) -> Unit,
    compact: Boolean
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .tppDropIn()
            .tppCard3D(shape = RoundedCornerShape(16.dp), elevation = 8.dp, lift = -2f)
            .background(BgPanel, RoundedCornerShape(16.dp))
            .border(1.dp, Line, RoundedCornerShape(16.dp))
            .padding(horizontal = 14.dp, vertical = 14.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        BrandBlock(compact = compact)
        TppTabBar(
            currentScreen = currentScreen,
            onScreenSelected = onScreenSelected,
            modifier = Modifier.fillMaxWidth()
        )
    }
}

@Composable
private fun BrandBlock(compact: Boolean) {
    val responsive = rememberResponsive()

    Box(
        modifier = Modifier.fillMaxWidth(),
        contentAlignment = Alignment.Center
    ) {
        TppLogo(
            height = responsive.logoHeight,
            maxWidth = responsive.logoMaxWidth
        )
    }
}

File: lib/screens/home_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../data/phone.dart';
import '../state/phone_controller.dart';
import '../theme/colors.dart';
import 'add_phone_screen.dart';
import 'budget_screen.dart';
import 'dashboard_screen.dart';
import 'inventory_screen.dart';
import 'phone_preview_sheet.dart';

class HomeScreen extends StatefulWidget {
  const HomeScreen({super.key});

  @override
  State<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  String? _shownToast;
  String? _shownError;
  String? _shownPreviewId;
  final _scaffoldKey = GlobalKey<ScaffoldState>();

  void _select(PhoneController controller, AppScreen screen) {
    final keepForm =
        screen == AppScreen.add && controller.editingPhone != null;
    controller.navigateTo(screen, keepForm: keepForm);
  }

  @override
  Widget build(BuildContext context) {
    final controller = context.watch<PhoneController>();

    WidgetsBinding.instance.addPostFrameCallback((_) {
      _handleSideEffects(controller);
    });

    return Scaffold(
      key: _scaffoldKey,
      drawer: _SideBar(
        current: controller.currentScreen,
        onSelect: (screen) {
          Navigator.of(context).pop();
          _select(controller, screen);
        },
      ),
      body: Container(
        decoration: const BoxDecoration(
          color: TppColors.bg,
          gradient: RadialGradient(
            center: Alignment(-0.7, -1),
            radius: 1.2,
            colors: [Color(0x145EEAD4), Colors.transparent],
          ),
        ),
        child: Material(
          type: MaterialType.transparency,
          child: SafeArea(
            child: Center(
              child: ConstrainedBox(
                constraints: const BoxConstraints(maxWidth: 1280),
                child: Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 16),
                  child: Column(
                    children: [
                      const SizedBox(height: 8),
                      _TopBar(
                        current: controller.currentScreen,
                        onMenu: () => _scaffoldKey.currentState?.openDrawer(),
                        onSelect: (screen) => _select(controller, screen),
                      ),
                      const SizedBox(height: 16),
                      Expanded(
                        child: AnimatedSwitcher(
                          duration: const Duration(milliseconds: 250),
                          child: KeyedSubtree(
                            key: ValueKey(controller.currentScreen),
                            child: _body(controller),
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }

  Widget _body(PhoneController controller) {
    if (controller.isLoading) {
      return const Center(
        child: CircularProgressIndicator(color: TppColors.accent),
      );
    }

    switch (controller.currentScreen) {
      case AppScreen.dashboard:
        return DashboardScreen(
          phones: controller.phones,
          totals: controller.totals,
          onOpenPhone: controller.openPreview,
        );
      case AppScreen.budget:
        return BudgetScreen(phones: controller.phones);
      case AppScreen.add:
        return AddPhoneScreen(
          editingPhone: controller.editingPhone,
          onSave: controller.savePhone,
          onClear: controller.resetForm,
          onCancelEdit: controller.resetForm,
        );
      case AppScreen.inventory:
        return InventoryScreen(
          controller: controller,
          onDelete: (phone) => _confirmDelete(controller, phone),
        );
    }
  }

  void _handleSideEffects(PhoneController controller) {
    final toast = controller.toastMessage;
    if (toast != null && toast != _shownToast) {
      _shownToast = toast;
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          content: Text(toast),
          backgroundColor: TppColors.bgElevated,
        ),
      );
      controller.clearToast();
    }

    final error = controller.errorMessage;
    if (error != null && error != _shownError) {
      _shownError = error;
      showDialog<void>(
        context: context,
        builder: (context) => AlertDialog(
          backgroundColor: TppColors.bgPanel,
          title: const Text('Error'),
          content: Text(error),
          actions: [
            TextButton(
              onPressed: () {
                Navigator.pop(context);
                controller.clearError();
              },
              child: const Text('OK'),
            ),
          ],
        ),
      );
    }

    final preview = controller.previewPhone;
    if (preview != null && preview.id != _shownPreviewId) {
      _shownPreviewId = preview.id;
      showPhonePreviewSheet(
        context: context,
        phone: preview,
        onEdit: () => controller.openEditForm(preview),
        onDelete: () => _confirmDelete(controller, preview),
      ).whenComplete(() {
        if (controller.previewPhone?.id == preview.id) {
          controller.closePreview();
        }
        _shownPreviewId = null;
      });
    }
  }

  Future<void> _confirmDelete(PhoneController controller, Phone phone) async {
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (context) => AlertDialog(
        backgroundColor: TppColors.bgPanel,
        title: const Text('Delete phone?'),
        content: Text('Remove ${phone.model} from inventory?'),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context, false),
            child: const Text('Cancel'),
          ),
          TextButton(
            onPressed: () => Navigator.pop(context, true),
            style: TextButton.styleFrom(foregroundColor: TppColors.danger),
            child: const Text('Delete'),
          ),
        ],
      ),
    );
    if (confirmed == true) {
      await controller.deletePhone(phone);
    }
  }
}

class _TopBar extends StatelessWidget {
  const _TopBar({
    required this.current,
    required this.onSelect,
    required this.onMenu,
  });

  final AppScreen current;
  final ValueChanged<AppScreen> onSelect;
  final VoidCallback onMenu;

  @override
  Widget build(BuildContext context) {
    final isNarrow = MediaQuery.sizeOf(context).width < 520;

    final menuButton = _MenuButton(onTap: onMenu);

    final brand = Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        menuButton,
        const SizedBox(width: 8),
        Container(
          width: 36,
          height: 36,
          decoration: BoxDecoration(
            shape: BoxShape.circle,
            gradient: const LinearGradient(
              colors: [TppColors.accentGradientStart, TppColors.accentGradientEnd],
            ),
            boxShadow: [
              BoxShadow(
                color: TppColors.glowRing.withValues(alpha: 0.3),
                blurRadius: 12,
              ),
            ],
          ),
          child: const Icon(Icons.public, color: TppColors.accentOn, size: 20),
        ),
        const SizedBox(width: 10),
        const Text(
          'TPP',
          style: TextStyle(
            fontSize: 22,
            fontWeight: FontWeight.w800,
            letterSpacing: 0.5,
          ),
        ),
      ],
    );

    if (isNarrow) {
      return Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          Align(alignment: Alignment.centerLeft, child: brand),
          const SizedBox(height: 12),
          _TabBar(current: current, onSelect: onSelect, expanded: true),
        ],
      );
    }

    return Row(
      children: [
        brand,
        const Spacer(),
        Flexible(
          child: _TabBar(current: current, onSelect: onSelect),
        ),
      ],
    );
  }
}

class _TabBar extends StatelessWidget {
  const _TabBar({
    required this.current,
    required this.onSelect,
    this.expanded = false,
  });

  final AppScreen current;
  final ValueChanged<AppScreen> onSelect;
  final bool expanded;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.all(4),
      decoration: BoxDecoration(
        color: TppColors.bgElevated,
        borderRadius: BorderRadius.circular(999),
        border: Border.all(color: TppColors.line),
      ),
      child: Row(
        mainAxisSize: expanded ? MainAxisSize.max : MainAxisSize.min,
        children: [
          _tab('Dashboard', AppScreen.dashboard),
          _tab('Add', AppScreen.add),
          _tab('Inventory', AppScreen.inventory),
        ],
      ),
    );
  }

  Widget _tab(String label, AppScreen screen) {
    final selected = current == screen;
    final tab = GestureDetector(
      onTap: () => onSelect(screen),
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 180),
        padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 8),
        decoration: BoxDecoration(
          color: selected ? TppColors.accentDim : Colors.transparent,
          borderRadius: BorderRadius.circular(999),
        ),
        child: Text(
          label,
          textAlign: TextAlign.center,
          maxLines: 1,
          overflow: TextOverflow.ellipsis,
          style: TextStyle(
            color: selected ? TppColors.accent : TppColors.textSecondary,
            fontWeight: FontWeight.w700,
            fontSize: 13,
          ),
        ),
      ),
    );
    return expanded ? Expanded(child: tab) : Flexible(child: tab);
  }
}

class _MenuButton extends StatelessWidget {
  const _MenuButton({required this.onTap});

  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    return Material(
      color: TppColors.bgElevated,
      borderRadius: BorderRadius.circular(12),
      child: InkWell(
        onTap: onTap,
        borderRadius: BorderRadius.circular(12),
        child: Container(
          width: 40,
          height: 40,
          alignment: Alignment.center,
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(12),
            border: Border.all(color: TppColors.line),
          ),
          child: const Icon(Icons.menu, size: 20, color: TppColors.textSecondary),
        ),
      ),
    );
  }
}

class _SideBar extends StatelessWidget {
  const _SideBar({required this.current, required this.onSelect});

  final AppScreen current;
  final ValueChanged<AppScreen> onSelect;

  static const _items = [
    (AppScreen.dashboard, 'Dashboard', Icons.dashboard_outlined),
    (AppScreen.budget, 'Budget', Icons.account_balance_wallet_outlined),
    (AppScreen.add, 'Add Phone', Icons.add_circle_outline),
    (AppScreen.inventory, 'Inventory', Icons.inventory_2_outlined),
  ];

  @override
  Widget build(BuildContext context) {
    return Drawer(
      backgroundColor: TppColors.bgPanel,
      shape: const RoundedRectangleBorder(
        borderRadius: BorderRadius.horizontal(right: Radius.circular(20)),
      ),
      child: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            Padding(
              padding: const EdgeInsets.fromLTRB(20, 24, 20, 20),
              child: Row(
                children: [
                  Container(
                    width: 42,
                    height: 42,
                    decoration: BoxDecoration(
                      shape: BoxShape.circle,
                      gradient: const LinearGradient(
                        colors: [
                          TppColors.accentGradientStart,
                          TppColors.accentGradientEnd,
                        ],
                      ),
                      boxShadow: [
                        BoxShadow(
                          color: TppColors.glowRing.withValues(alpha: 0.3),
                          blurRadius: 12,
                        ),
                      ],
                    ),
                    child: const Icon(Icons.public,
                        color: TppColors.accentOn, size: 22),
                  ),
                  const SizedBox(width: 12),
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    mainAxisSize: MainAxisSize.min,
                    children: const [
                      Text(
                        'TPP',
                        style: TextStyle(
                          fontSize: 20,
                          fontWeight: FontWeight.w800,
                          letterSpacing: 0.5,
                        ),
                      ),
                      Text(
                        "The Phone's Planet",
                        style: TextStyle(color: TppColors.muted, fontSize: 12),
                      ),
                    ],
                  ),
                ],
              ),
            ),
            const Divider(color: TppColors.line, height: 1),
            const SizedBox(height: 8),
            for (final item in _items)
              _SideBarItem(
                label: item.$2,
                icon: item.$3,
                selected: current == item.$1,
                onTap: () => onSelect(item.$1),
              ),
            const Spacer(),
            const Padding(
              padding: EdgeInsets.all(20),
              child: Text(
                'Inventory & Budget',
                style: TextStyle(color: TppColors.muted2, fontSize: 11),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class _SideBarItem extends StatelessWidget {
  const _SideBarItem({
    required this.label,
    required this.icon,
    required this.selected,
    required this.onTap,
  });

  final String label;
  final IconData icon;
  final bool selected;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 3),
      child: Material(
        color: selected ? TppColors.accentDim : Colors.transparent,
        borderRadius: BorderRadius.circular(12),
        child: InkWell(
          onTap: onTap,
          borderRadius: BorderRadius.circular(12),
          child: Container(
            padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 12),
            decoration: BoxDecoration(
              borderRadius: BorderRadius.circular(12),
              border: Border.all(
                color: selected
                    ? TppColors.accent.withValues(alpha: 0.4)
                    : Colors.transparent,
              ),
            ),
            child: Row(
              children: [
                Icon(
                  icon,
                  size: 20,
                  color: selected ? TppColors.accent : TppColors.textSecondary,
                ),
                const SizedBox(width: 14),
                Text(
                  label,
                  style: TextStyle(
                    color:
                        selected ? TppColors.accent : TppColors.textSecondary,
                    fontWeight: FontWeight.w700,
                    fontSize: 15,
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
